main.cpp

```diff
@@ -243,8 +206,26 @@ struct IntType
     IntType& pow(const DoubleType&);
     IntType& pow(int);
     
+    IntType& apply(std::function<IntType&(std::unique_ptr<int>&)> IntFptr)
+    {
+        if (IntFptr)
+        {
+            return IntFptr(value);
+        }
+        return *this;
+    }
+    
+    IntType& apply(void(*IntFptr)(std::unique_ptr<int>&))
```

Comment:
Follow the coding standard in the course.
Variable names need to start with a lowercase letter.
- [ ] revise